iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/ShuntCompensatorAdderImpl.java

```diff
@@ -177,6 +180,17 @@ public ShuntCompensatorAdder setTargetDeadband(double targetDeadband) {
     public ShuntCompensatorImpl add() {
         String id = checkAndGetUniqueId();
         TerminalExt terminal = checkAndGetTerminal();
+        ShuntCompensatorModelWrapper model = null;
+        if (shuntCompensatorLinearModelAdder != null) {
+            model = new ShuntCompensatorLinearModelImpl(shuntCompensatorLinearModelAdder.bPerSection,
+                                                        shuntCompensatorLinearModelAdder.gPerSection,
+                                                        shuntCompensatorLinearModelAdder.maximumSectionCount);
+        } else if (shuntCompensatorNonLinearModelAdder != null) {
+            model = new ShuntCompensatorNonLinearModelImpl(shuntCompensatorNonLinearModelAdder.sections.stream()
+                                                                                                       .map(ShuntCompensatorNonLinearModelImpl.SectionImpl::copy)
+                                                                                                       .collect(Collectors.toList()));
```

Comment:
I would perform the deep copy in the constructor of ShuntCompensatorNonLinearModelImpl (and document it)